src/index.tsx

```diff
@@ -10,125 +10,125 @@ function getValue(target: any, { isCheckbox }: { isCheckbox: boolean }) {
+const RHFInput = <T,>({
+  innerProps,
+  setValue: setValueFromProp,
+  register: registerFromProp,
+  unregister: unregisterFromProp,
+  name,
+  rules,
+  mode = 'onSubmit',
+  as: InnerComponent,
+  onChange,
+  onBlur,
+  type,
+  value,
+  defaultValue,
+  defaultChecked,
+  onChangeName,
+  onChangeEvent,
+  onBlurName,
+  onBlurEvent,
+  ...rest
+}: Props<T>) => {
+  const isCheckbox = type === 'checkbox';
+  const isOnChange = mode === 'onChange';
+  const isOnBlur = mode === 'onBlur';
+  const defaultData = isCheckbox
+    ? defaultChecked === undefined
+      ? false
+      : defaultChecked
+    : defaultValue === undefined
+    ? ''
+    : defaultValue;
+  const [inputValue, setInputValue] = React.useState(defaultData);
+  const valueRef = React.useRef(defaultData);
+  const methods = useFormContext() || {};
+  const setValue = setValueFromProp || methods.setValue;
+  const register = registerFromProp || methods.register;
+  const unregister = unregisterFromProp || methods.unregister;
 
-    const commonTask = (target: any) => {
-      const data = getValue(target, { isCheckbox });
-      setInputValue(data);
-      valueRef.current = data;
-      return data;
-    };
+  const commonTask = (target: any) => {
+    const data = getValue(target, { isCheckbox });
+    setInputValue(data);
+    valueRef.current = data;
+    return data;
+  };
 
-    const eventWrapper = (event: EventFunction, eventName: string) => {
-      return (...arg: any) => {
-        const data = commonTask(event(arg));
-        setValue(
-          name,
-          data,
-          (isOnChange && eventName === 'onChange') ||
-            (isOnBlur && eventName === 'onBlur'),
-        );
-      };
+  const eventWrapper = (event: EventFunction, eventName: string) => {
+    return (...arg: any) => {
+      const data = commonTask(event(arg));
+      setValue(
+        name,
+        data,
+        (isOnChange && eventName === 'onChange') ||
+          (isOnBlur && eventName === 'onBlur'),
+      );
     };
+  };
 
-    const handleChange = (e: any) => {
-      const data = commonTask(e && e.target ? e.target : e);
-      setValue(name, data, isOnChange);
-      if (onChange) {
-        onChange(e);
-      }
-    };
+  const handleChange = (e: any) => {
+    const data = commonTask(e && e.target ? e.target : e);
+    setValue(name, data, isOnChange);
+    if (onChange) {
+      onChange(e);
+    }
+  };
 
-    const handleBlur = (e: any) => {
-      const data = commonTask(e && e.target ? e.target : e);
-      setValue(name, data, isOnBlur);
-      if (onBlur) {
-        onBlur(e);
-      }
-    };
+  const handleBlur = (e: any) => {
+    const data = commonTask(e && e.target ? e.target : e);
+    setValue(name, data, isOnBlur);
+    if (onBlur) {
+      onBlur(e);
+    }
+  };
 
     React.useEffect(() => {
       register(
         Object.defineProperty(
           {
             name,
           },
-          'value',
-          {
-            set(data) {
```

Comment:
did you remove the set by accident?

Reply:
Oops. Sorry. Too many mistakes

Reply:
@JeromeDeLeon don't be silly, you are doing a great job! let's merge this if you can fix the below issue. (the screenshot)